editor/editor_dock_manager.cpp

```diff
@@ -345,14 +344,58 @@ void EditorDockManager::_move_dock(Control *p_dock, Control *p_target, int p_tab
 	p_target->set_block_signals(false);
 	TabContainer *dock_tab_container = Object::cast_to<TabContainer>(p_target);
 	if (dock_tab_container) {
-		dock_tab_container->set_tab_title(dock_tab_container->get_tab_idx_from_control(p_dock), all_docks[p_dock].title);
+		const DockInfo &dock_info = all_docks[p_dock];
+		const int target_index = dock_tab_container->get_tab_idx_from_control(p_dock);
+
+		if (dock_tab_container->is_inside_tree()) {
+			_update_tab_style(p_dock);
+		} else {
+			dock_tab_container->set_tab_title(target_index, dock_info.title);
+		}
```

Comment:
I wonder why was this here. Seems to be unnecessary.

Reply:
The `set_tab_title` seems unnecessary indeed. The inside tree part is mainly for cases like de-floating the dock, i.e., adding the dock control back to a TabContainer.